app/controllers/api/v1/locations_controller.rb

```
@@ -0,0 +1,99 @@
+    if coordinate_search?
+      search_results = LocationSearch::PointFinder.new(current_api_user, search_params).call
+
+      render json: Api::LocationSearchResultSerializer.new(search_results).call
+    else
+      render json: { error: 'Coordinates (lat, lon) are required' }, status: :bad_request
+    end
+  rescue StandardError => e
+    Rails.logger.error "Location search error: #{e.message}"
+    Rails.logger.error e.backtrace.join("\n")
+    render json: { error: 'Search failed. Please try again.' }, status: :internal_server_error
+  end
```

RuboCop (1.76.1)

[convention] Assignment Branch Condition size for `index` is too high. [<2, 17, 3> 17.38/17]

(Metrics/AbcSize)